tips/TIP-0019/tip-0019.md

```diff
@@ -0,0 +1,2950 @@
+
+To prevent Bob from blocking access to the `storage deposit` forever, Alice specifies the additional `Expiration Unlock Condition` in the `Basic Output`. If Bob does not consume the output before the time window defined by Alice expires, Alice regains total control over the output.
+
+This means that there is no risk for Alice to loose the `storage deposit`, because either Bob needs to return the specified `Return Amount`, or the ownership of the created output switches back to Alice after the specified time-window has expired.
+
+This mechanism can also be used to transfer native tokens or on-chain requests to ISCP chains without losing control over the required `storage deposit`.
+
+#### Microtransactions on Layer 2
+
+Another solution is to outsource microtransactions to Layer 2 applications like smart contracts. In Layer 2 there are no restrictions regarding the minimum balance of an output.
+
+![Microtransactions on Layer 2](assets/microtransactions_pt3_layer2.png)
+
+In this example, Alice sends funds to a smart contract chain on Layer 1 with an output that covers at least `min_deposit_of_output`. From this point on, Alice can send any number of off-chain requests to the smart contract chain, causing the smart contract to send microtransactions from Alice' on-chain account to Bob's on-chain account. Bob can now request his on-chain account balances to be withdrawn to his Layer 1 address. The last step can also be combined with the former introduced `conditional sending` mechanism, in case Bob wants to withdraw less than `min_deposit_of_output` IOTA coins or native assets.
+
+| :information_source: Potential additional mechanisms for microtransactions are currently being discussed. |
+| --------------------------------------------------------------------------------------------------------- |
+
+### Migration from old to new dust protection
+
+All `SigLockedSingleOutput` below 1 MIOTA and `SigLockedDustAllowanceOutput` of an address could be collected and migrated to a single new `BasicOutput` with the smallest `Output ID` (byte-wise) of all these collected outputs as the new identifier.
+
+This could probably be done in the form of a global snapshot and would represent a hard-fork.
+
+Another solution is to convert all `SigLockedDustAllowanceOutput` into `BasicOutputs` and leave the `SigLockedSingleOutput` below 1 MIOTA untouched.
+
+### Unresolved questions
```

Comment:
We should remove this section as you said because any parameter values are subject to be put into a separate TIP or the actual network configuration anyway, so this is not the place to put it.

Reply:
I will removed it :+1: